src/Stream.php

```diff
@@ -89,6 +89,29 @@ public static function create($body = ''): StreamInterface
         throw new \InvalidArgumentException('First argument to Stream::create() must be a string, resource or StreamInterface.');
     }
 
+    /**
+     * Creates a new PSR-7 stream from path
+     *
+     * @internal Implements Psr\Http\Message\StreamFactoryInterface::createStreamFromFile for internal use, library users should default to the actual Factory object.
+     */
+    public static function createFromFile(string $filename, string $mode = 'r'): StreamInterface
+    {
+        try {
+            $resource = @\fopen($filename, $mode);
+        } catch (\Throwable $e) {
+            throw new \RuntimeException('The file ' . $filename . ' cannot be opened.');
+        }
+        if (false === $resource) {
+            if ('' === $mode || false === \in_array($mode[0], ['r', 'w', 'a', 'x', 'c'])) {
```

Comment:
strict in_array plz

Reply:
Done. Is this faster internally? Because this check is always comparing between same types by virtue of its arguments.